### PR TITLE
Release SegmentGroup reference count

### PR DESCRIPTION
**Pull request by chaoyli**
In streaming ingestion, segment group is set to be one in creation.
Upon closing, reference count should to be released. Otherwise,
file descriptor and segment group object in memory can not be freed.

### CODE REVIEW

**Comment by chaoyli** [comment]
associated issue is #690

File: be/src/olap/delta_writer.cpp [hunk 1]
```diff
@@ -38,6 +38,9 @@ DeltaWriter::~DeltaWriter() {
     if (!_delta_written_success) {
         _garbage_collection();
     }
+    for (SegmentGroup* segment_group : _segment_group_vec) {
+        segment_group->release();
+    }
```
Comment (imay) [inline be/src/olap/delta_writer.cpp]: `_garbage_collection` and this code will release a segment group twice

Comment (chaoyli) [reply be/src/olap/delta_writer.cpp]: OK

**Comment by kangkaisen** [comment]
Do we still need to release `segment_group` in `OLAPTable::delete_pending_data` ?

**Comment by chaoyli** [comment]
> Do we still need to release `segment_group` in `OLAPTable::delete_pending_data` ?

If delta_writer has release segment group, OLAPTable::delete_pending_data is not necessary to release it. When add_pending_data to olap_table, reference count of SegmentGroup is not added by one.
This is may be trick, and will be refactored in the following release.

**Comment by kangkaisen** [comment]
@chaoyli OK.

File: be/src/olap/delta_writer.cpp (hunk 1 above)
Comment (imay) [reply be/src/olap/delta_writer.cpp]: why `_new_segment_group_vec` doesn't need release?

**Review by imay — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**